pkgs/build-support/make-darwin-application/default.nix

```diff
@@ -0,0 +1,59 @@
+# given a package with an executable and an icon, make a darwin application for it.
+
+{ lib, writeShellScript }:
+
+{ name # The name of the Application file
+, exec
+, icon ? null
+, version
+}:
+
+let
+  plist = {
+    "CFBundleDevelopmentRegion" = "English";
+    "CFBundleExecutable" = name;
+    "CFBundleIconFile" = "${name}.icns";
+    "CFBundleIdentifier" = "org.nixos.${name}";
+    "CFBundleInfoDictionaryVersion" = "6.0";
+    "CFBundleName" = name;
+    "CFBundlePackageType" = "APPL";
+    "CFBundleShortVersionString" = version;
+    "CFBundleSignature" = "???";
```

Comment:
Does this mean the bundle is never signed and is there an impact on performance because of this? IIRC from Catalina forward there's a startup penalty for unsigned executables?

Reply:
Ah, I didn't know that.  Is there a way that we can calculate this signature?